src/pydicom/dataset.py

```diff
@@ -2656,6 +2660,56 @@ def ensure_file_meta(self) -> None:
         if not hasattr(self, "file_meta"):
             self.file_meta = FileMetaDataset()
 
+    def validate(self, _init_value: _DatasetType) -> None:
+        """Validate the dataset.
+
+        Performs the same validation checks as dcmwrite without writing to disk.
```

Comment:
I'm not sure about which `dcmwrite` argument(s) are you referring to here? Most of them seem to be write-specific, no?

Reply:
Ah well, I thought that there is quite a bit of shared code that could be factored out, but as the exception messages are also different, it may not be worth it.